### ISSUE TITLE
BPR loss sigmoid vs softplus

### ISSUE THREAD

**Issue by YuZhang10** [gusye1234/pytorch-light-gcn#37]
Hello, thanks for your implementation in the first place.
I notice that in BPR loss you used **softplus** instand of **sigmiod**, which is different from the original paper. 
![image](https://user-images.githubusercontent.com/30391848/228281691-f4877189-27d9-4430-9b2f-c6c92cec5935.png)
Could you please explain why do so? Thx~

**Comment by zbt78** [comment]
They are same!
If we use sigmoid, BPRloss = -logσ(pos-neg) = -log(1 / (1+e^-(pos - neg))).
If we use softplus, BPRloss = softplus(neg - pos) = log(1+e^(neg -  pos)) = -log(1 / (1+e^-(pos - neg))).
they are equal.
sigmoid(x) = 1 / (1 + e^(-x));     softplus(x) = log(1 + e^x) .

**Comment by zbt78** [comment]
They are same!
If we use sigmoid, BPRloss = -logσ(pos-neg) = -log(1 / (1+e^-(pos - neg))).
If we use softplus, BPRloss = softplus(neg - pos) = log(1+e^(neg -  pos)) = -log(1 / (1+e^-(pos - neg))).
they are equal.
sigmoid(x) = 1 / (1 + e^(-x));     softplus(x) = log(1 + e^x) .
